src/rules/caisson-id-check.js

```diff
@@ -0,0 +1,248 @@
+/**
+ * @title Caisson ID Check
+ * @overview Validate US driver's licenses and international passports in real time.
+ * @gallery true
+ * @category marketplace
+ *
+ * [Caisson Integration guide](https://www.caisson.com/docs/integration/auth0/)
+ */
+
+async function caissonIDCheck(user, context, callback) {
+  const { Auth0RedirectRuleUtilities } = require("@auth0/rule-utilities@0.1.0");
+
+  //copy off the config obj so we can use our own private key for session token signing.
+  let caissonConf = JSON.parse(JSON.stringify(configuration));
+  caissonConf.SESSION_TOKEN_SECRET = configuration.CAISSON_PRIVATE_KEY;
+
+  const manager = {
+    creds: {
+      public_key: caissonConf.CAISSON_PUBLIC_KEY,
+      private_key: caissonConf.CAISSON_PRIVATE_KEY,
+    },
+    debug:
+      caissonConf.CAISSON_DEBUG &&
+      caissonConf.CAISSON_DEBUG.toLowerCase() === "true"
+        ? true
+        : false,
+    idCheckFlags: {
+      login_frequency_days: parseInt(caissonConf.CAISSON_LOGIN_FREQUENCY_DAYS),
+    },
+    caissonHosts: {
+      idcheck: "https://id.caisson.com",
+      api: "https://api.caisson.com",
+      dashboard: "https://www.caisson.com",
+    },
+    axios: require("axios"),
+    util: new Auth0RedirectRuleUtilities(user, context, caissonConf),
+  };
+
+  /**
+   * Toggleable logger.  Set CAISSON_DEBUG in the Auth0 configuration to enable.
+   *
+   * @param {error} err
+   */
+  function dLog(err) {
+    if (manager.debug) {
+      console.log(err);
+    }
+  }
+
+  /**
+   * Helper function for converting milliseconds to days. Results rounded down.
+   * @param {int} mils
+   */
+  function millisToDays(mils) {
+    return Math.floor(mils / 1000 / 60 / 60 / 24);
+  }
+
+  /**
+   * Creates Caisson specific session token and sets redirect url.
+   */
+  function setIDCheckRedirect() {
+    const token = manager.util.createSessionToken({
+      public_key: manager.creds.public_key,
+      host: context.request.hostname,
+      user_id: user.user_id,
```

Comment:
The user ID is already added as `sub` in there, just FYI

Reply:
moved to using "sub"